HW0/P4/P4.py

```diff
@@ -2,6 +2,73 @@
 import matplotlib.pyplot as plt
 from mpl_toolkits.mplot3d import Axes3D
 
+def predictS(prop, sk, a):
+    """Implements Eq. 2 in the document
+    Returns s~
+    prop is the propagation matrix
+    sk is the current vector
+    a is the "gravity vector"
+    """
+
+    return np.dot(prop, sk) + a
+
+# Implemenets Eq. 3 in the document
+# Returns Sigma~
+def predictSig(prop, sigk, B):
+    """Implemenets Eq. 3 in the document
+    Returns Sigma~
+    prop is the propagation matrix
+    sigk is our curent sigma
+    B is the B matrix"""
+
+    # Compute the first term for clarity
+    first_term = np.dot(np.dot(prop, sigk), np.transpose(A))
+    
+    # Compute the second term for clarity
+    second_term = np.dot(B, np.transpose(B))
+
+    # Combine for the parenthetical term
+    paren = first_term + second_term
+
+    # Invert and return sigma twiddle
+    return np.linalg.inv(paren)
+
+
+def updateSig(Sig_twid, C):
```

Comment:
Try to avoid using uppercase variable names like `Sig_twid`, as these are generally reserved for class names in Python (see [PEP8](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables))